### ISSUE TITLE
Authentication scripts not passing requests to browser

### ISSUE THREAD

**Issue by benzionyunger** [zaproxy/zaproxy#5609]
**Describe the bug**
Authentication is only performing via the proxy, but the session is not being passed to the browser, therefore not populating the site tree with the URLs.
**To Reproduce**
1. Zest/JS scripts perform authentication correctly when running in the script console, the Zest Results show all GET and POST requests with the login info, token validation, and HTTP Sender for updating all requests with the authorization token.
2. Loading the script to the context and running a spider+ajax spider with the context on a fresh site tree.
3. Output tab shows authorization successful, but history shows only the last GET request of the script to access the site's dashboard. No other auth requests are displayed in the History tab. And Output tab also displays auth successful. Even the script request in the History tab is not populated on the site-tree.
4. Spider populates site tree with attempted logins with false data, but does not perform a successful login. However, all those requests were made with an auth token which matches the auth token printed to the scripts console while the script is running. (The responses of those requests were rejected by the server)

It seems, that the proxy performs an authorization, then passes the token to the browser with a new session.
**Expected behavior**
I expected the script to run from the browser and populate the site tree with a POST to a successful login and GET request to the dashboard.
**Screenshots**

**Software versions**
 - ZAP: 2.8.0
 - OS: Ubuntu 18.04
 - Java: openjdk version "11.0.4"
 - Browser:Zap built in firefox.

**Errors from the zap.log file**
See https://github.com/zaproxy/zaproxy/wiki/FAQconfig for the location of the zap.log file for your OS.

**Additional context**
I attempted the authorization from JSON-based authorization and then from the scripts, both had the same results. I also tried with and without a 'user' and 'force user' settings.
Another point to note, the http session tab never displays any sessions during any runs.

**Comment by psiinon** [comment]
@benzionyunger this is a known restriction, hence the change in title - its definitely something we want to address.

**Comment by benzionyunger** [comment]
Is there any workaround? 
Would first crawling the site with a selenium driver resolve my issue?

**Comment by thc202** [comment]
Yes, that would work (that would replace the need for using the AJAX Spider).

**Comment by psiinon** [comment]
FYI this can be now acheived via Selenium Scripts https://github.com/zaproxy/community-scripts/tree/master/selenium but leaving this open as we could still provide an easier to configure solution...

**Comment by psiinon** [comment]
Browser based auth makes this even easiier, but we do need to provide better integration with the AJAX Spider: https://www.zaproxy.org/docs/desktop/addons/authentication-helper/browser-auth/

**Comment by psiinon** [comment]
We now have proper integration with the AJAX Spider.
We can pass authentication state to selected apps, but only via scripts. I've not found any generic way to do this, so I think scripts will remain the only option in this case.
But browser based auth is the best option where that works.
